app/assets/javascripts/Components/Modals/submit_view_token_modal.jsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import {render} from "react-dom";
+import Modal from "react-modal";
+
+class SubmitViewTokenModal extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isOpen: false,
+      token: null,
+    };
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+  }
+
+  onSubmit = () => {
+    $.ajax({
+      url: Routes.view_token_check_course_result_path(this.props.course_id, this.props.result_id, {
+        view_token: this.state.token,
+      }),
+    }).then(
+      () => {
+        window.location = Routes.view_marks_course_result_path(
+          this.props.course_id,
+          this.props.result_id,
+          {view_token: this.state.token}
+        );
+      },
+      () => this.setState({isOpen: false, token: null})
+    );
+  };
+
+  render() {
+    return (
+      <Modal
+        className="react-modal"
+        isOpen={this.state.isOpen}
+        onRequestClose={() => this.setState({isOpen: false, token: null})}
+      >
+        <p>{I18n.t("results.view_token_submit")}</p>
+        <form onSubmit={this.onSubmit}>
+          <div className={"modal-container-vertical"}>
+            <div className={"modal-container"}>
+              <input onChange={e => this.setState({token: e.target.value})} />
+            </div>
+            <div className={"modal-container"}>
+              <input type="submit" value={I18n.t("save")} />
```

Comment:
Please introduce a new string label for this button. Something like "Continue" or "Submit" or "Submit Token".